lms/services/vitalsource/factory.py

```diff
@@ -5,9 +5,13 @@
 def service_factory(_context, request):
     settings = request.find_service(name="application_instance").get_current().settings
 
+    customer_key = settings.get("vitalsource", "api_key")
+
     return VitalSourceService(
-        client=VitalSourceClient(
-            api_key=request.registry.settings["vitalsource_api_key"]
-        ),
+        # It's important to pass None here if there's no key, so the service
+        # knows it's been disabled. The client will raise an error anyway if
+        # you try and create one with no API key.
+        client=VitalSourceClient(customer_key) if customer_key else None,
+        user_lti_param=settings.get("vitalsource", "user_lti_param"),
```

Comment:
We could do `user_reference=request.lti_params[settings.get("vitalsource", "user_lti_param")]`

That simplifies a bit the usage but also makes the service only usable on LTILaunches so the current setup is a good choice too

Reply:
This hadn't occurred to me. I wonder if this would be confusing in methods like `get_launch_url` where you accept `user_reference` as an argument? In that case it would actually be blank when called (as it's coming from the API, not an LTI launch), but it would be reasonable to ask why it's not using `self.user_reference` if it's around.

We could make it work, by calling the API with the param listed so they always line up, but I feel like that's actually quite a weird way to follow how the data got in.

I suspect we might run into a problem with this even as is, if people actually do use really different values for this, as anything which falls outside of the LTI params we look for would require us to use `request.params`, or add them and at that point it would lose all LTI 1.3 compatibility.